src/Symfony/Component/Intl/Data/Generator/CurrencyDataGenerator.php

```diff
@@ -156,4 +221,41 @@ private function generateNumericToAlpha3Mapping(array $alpha3ToNumericMapping):
 
         return $numericToAlpha3Mapping;
     }
+
+    /**
+     * Decodes ICU "date pair" into a DateTimeImmutable (UTC).
+     *
+     * ICU stores UDate = milliseconds since 1970-01-01T00:00:00Z in a signed 64-bit.
+     *
+     * @param array{0: int, 1: int} $pair
+     */
+    private static function icuPairToDate(array $pair): string
+    {
+        [$highBits32, $lowBits32] = $pair;
+
+        // Recompose a 64-bit unsigned integer from two 32-bit chunks.
+        $unsigned64 = ((($highBits32 & 0xFFFFFFFF) << 32) | ($lowBits32 & 0xFFFFFFFF));
+
+        // Convert to signed 64-bit (two's complement) if sign bit is set.
+        if ($unsigned64 >= (1 << 63)) {
+            $unsigned64 -= (1 << 64);
+        }
```

Comment:
The psalm [job fails](https://github.com/symfony/symfony/actions/runs/17387649756/job/49356615487?pr=61583) since this commit due to a bug: https://github.com/vimeo/psalm/issues/11209

Reply:
This condition is not necessary and the fix is not working. It can be removed: https://github.com/symfony/symfony/pull/61681